src/app/(nextjs_migration)/(guest)/layout.tsx

```diff
@@ -2,23 +2,70 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import { v5 as uuidv5 } from "uuid";
 import { ReactNode } from "react";
 import "../../../client/css/index.css";
 import Image from "next/image";
 import MonitorLogo from "../../../client/images/monitor-logo-transparent@2x.webp";
 import MozillaLogo from "../../../client/images/moz-logo-1color-white-rgb-01.svg";
 import { SignInButton } from "../components/client/SignInButton";
 import { getL10n } from "../../functions/server/l10n";
+import { getServerSession } from "next-auth";
+import { authOptions } from "../../api/utils/auth";
+import { cookies } from "next/headers";
+import { randomUUID } from "crypto";
+import { PageLoadEvent } from "../components/client/PageLoadEvent";
+import { getExperiments } from "../../functions/server/getExperiments";
 
 export type Props = {
   children: ReactNode;
 };
 
-const GuestLayout = (props: Props) => {
+const GuestLayout = async (props: Props) => {
   const l10n = getL10n();
 
+  // If the user is logged in, use UUID derived from FxA UID as Nimbus user ID.
+  const session = await getServerSession(authOptions);
+  const accountId = session?.user?.subscriber?.fxa_uid;
+  let userId = "";
+
+  if (accountId && typeof accountId === "string") {
+    // If the user is logged in, use a UUID based on the user's subscriber ID.
+    // Note: we may want to use the FxA UID here, but we need approval for that first.
+    if (process.env.NIMBUS_UUID_NAMESPACE) {
+      userId = uuidv5(accountId, process.env.NIMBUS_UUID_NAMESPACE);
+    } else {
+      console.error("NIMBUS_UUID_NAMESPACE env var not set");
+    }
```

Comment:
Not sure if this is helpful or makes sense, but my eyes are telling me this might be very close (or copypaste) of src/app/(nextjs_migration)/(authenticated)/user/layout.tsx#L33-44.
I dont know if it makes sense to abstract that out, or if that is just overoptimizing.

I think the only differences between the files/blocks is that this has an `else` block while the authenticated layout doesn't.

```js
function userIdFromSession(session) {
  const accountId = session?.user?.subscriber?.fxa_uid;
  let userId = "";

  if (accountId && typeof accountId === "string") {
    // If the user is logged in, use a UUID based on the user's subscriber ID.
    // Note: we may want to use the FxA UID here, but we need approval for that first.
    if (process.env.NIMBUS_UUID_NAMESPACE) {
      userId = uuidv5(accountId, process.env.NIMBUS_UUID_NAMESPACE);
    } else {
      console.error("NIMBUS_UUID_NAMESPACE env var not set");
    }
  }
  return userId;
}
```

which might change this to:
```js
let userId = userIdFromSession(session);
if (!userId) {
    // if the user is not logged in, use a cookie with a randomly-generated Nimbus user ID.
    const cookie = cookies().get("userId");
    if (cookie) {
      userId = cookie.value;
    } else {
      // TODO Cookies can only be set in server action or route handler
      // @see https://nextjs.org/docs/app/api-reference/functions/cookies#cookiessetname-value-options
      // cookies().set("userId", uuid);
      userId = `guest-${randomUUID()}`;
    }
}
```
I honestly don't know if that is any better, or if the code is slightly different.

Reply:
Yes they are pretty similar, but different enough I didn't feel like it was worth abstracting just that one piece - I also think that we'll probably change this if we get approval to use FxA UID (filed a ticket to track that)